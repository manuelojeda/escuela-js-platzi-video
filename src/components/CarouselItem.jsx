import React from 'react';
import '../assets/styles/components/CarouselItem.scss';

const CarouselItem = () => (
  <div className='carousel-item'>
    <img className='carousel-item__img' src='https://images.pexels.com/photos/789822/pexels-photo-789822.jpeg?auto=format%2Ccompress&cs=tinysrgb&dpr=2&h=750&w=1260' alt='' />
    <div className='carousel-item__details'>
      <div>
        {/* <img className='carousel-item__details--img' src='../assets/play-icon.png' alt='Play Icon' /> */}
        <img className='carousel-item__details--img' src='../assets/images/play-icon.png' alt='Play Icon' />
        <img className='carousel-item__details--img' src='../assets/images/plus-icon.png' alt='Plus Icon' />
      </div>
      <p className='carousel-item__details--title'>TÃ­tulo descriptivo</p>
      <p className='carousel-item__details--subtitle'>2019 16+ 114 minutos</p>
    </div>
  </div>
);

export default CarouselItem;
