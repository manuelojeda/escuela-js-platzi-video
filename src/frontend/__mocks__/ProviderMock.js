import React from 'react';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import initialState from '../initialState'
import { createBrowserHistory } from 'history';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)))
const history = createBrowserHistory();

const ProviderMock = (props) => (
  <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
  </Provider>
)

export default ProviderMock
