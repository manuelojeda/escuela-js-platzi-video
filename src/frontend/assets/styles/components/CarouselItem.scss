.carousel-item {
    border-radius: 20px;
    overflow: hidden;
    position: relative;
    display: inline-block;
    width: 200px;
    height: 250px;
    margin-right: 10px;
    font-size: 20px;
    cursor: pointer;
    transition: 450ms all;
    -webkit-transform-origin: center left;
            transform-origin: center left;
  }
  
  .carousel-item:hover ~ .carousel-item {
    -webkit-transform: translate3d(100px, 0, 0);
            transform: translate3d(100px, 0, 0);
  }
  
  .carousel-item__img {
    width: 200px;
    height: 250px;
    -o-object-fit: cover;
       object-fit: cover;
  }
  
  .carousel-item__details {
    align-items: flex-start;
    background: linear-gradient(to top, rgba(0,0,0,0.9) 0%, rgba(0,0,0,0) 100%);
    bottom: 0;
    display: flex;
    font-size: 10px;
    flex-direction: column;
    justify-content: flex-end;
    left: 0;
    opacity: 0;
    transition: 450ms opacity;
    padding: 10px;
    position: absolute;
    right: 0;
    top: 0;
  }
  
  .carousel-item__details--img {
    width: 20px;
  }
  
  .carousel-item:hover .carousel-item__details {
    opacity: 1;
  }
  
  .carousel-item__details--title {
    color: white;
    margin: 5px 0px 0px 0px;
  }
  
  .carousel-item__details--subtitle {
    color: white;
    font-size: 8px;
    margin: 3px 0px;
  }