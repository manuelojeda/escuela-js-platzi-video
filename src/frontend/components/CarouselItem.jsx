import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { setFavorite, unsetFavorite } from '../actions';
import '../assets/styles/components/CarouselItem.scss';
import Play from '../assets/images/play-icon.png';
import Plus from '../assets/images/plus-icon.png';
import Remove from '../assets/images/remove-icon.png';

const CarouselItem = (props) => {
  const { cover, title, year, contentRating, duration, id, isListBand } = props;

  const handleSetFavorite = () => {
    props.setFavorite({
      cover, title, year, contentRating, duration, id,
    });
  };

  const handleUnsetFavorite = (itemId) => {
    props.unsetFavorite(itemId);
  };

  return (
    <div className='carousel-item'>
      <img className='carousel-item__img' src={cover} alt={title} />
      <div className='carousel-item__details'>
        <div>
          <Link to={`/player/${id}`}>
            <img
              className='carousel-item__details--img'
              src={Play}
              alt='Play Icon'
            />
          </Link>
          {!isListBand && (
            <img
              className='carousel-item__details--img'
              src={Plus}
              alt='Plus Icon'
              onClick={handleSetFavorite}
            />
          )}
          {isListBand && (
            <img
              className='carousel-item__details--img'
              src={Remove}
              alt='Remove Icon'
              onClick={() => handleUnsetFavorite(id)}
            />
          )}
        </div>
        <p className='carousel-item__details--title'>{title}</p>
        <p className='carousel-item__details--subtitle'>
          {`${year} ${contentRating} ${duration} minutos`}
        </p>
      </div>
    </div>
  );
};

CarouselItem.propTypes = {
  cover: PropTypes.string,
  title: PropTypes.string,
  year: PropTypes.number,
  contentRating: PropTypes.string,
  duration: PropTypes.number,
};

const mapDispatchToProps = {
  setFavorite,
  unsetFavorite,
};

export default connect(null, mapDispatchToProps)(CarouselItem);
