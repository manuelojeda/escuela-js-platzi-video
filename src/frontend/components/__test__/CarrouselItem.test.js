import React from 'react'
import { create } from 'react-test-renderer'
import ProviderMock from '../../__mocks__/ProviderMock'
import CarouselItem from '../CarouselItem'

describe('CarouselItem testing', () => {
    test('Match snapshot', () => {
      const carrousel = create(
        <ProviderMock>
          <CarouselItem />
        </ProviderMock>
      )

      expect(carrousel.toJSON()).toMatchSnapshot();
    })
})
