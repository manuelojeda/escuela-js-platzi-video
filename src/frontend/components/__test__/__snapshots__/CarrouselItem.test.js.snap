// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselItem testing Match snapshot 1`] = `
<div
  className="carousel-item"
>
  <img
    className="carousel-item__img"
  />
  <div
    className="carousel-item__details"
  >
    <div>
      <a
        href="/player/undefined"
        onClick={[Function]}
      >
        <img
          alt="Play Icon"
          className="carousel-item__details--img"
          src="test-file-stubs"
        />
      </a>
      <img
        alt="Plus Icon"
        className="carousel-item__details--img"
        onClick={[Function]}
        src="test-file-stubs"
      />
    </div>
    <p
      className="carousel-item__details--title"
    />
    <p
      className="carousel-item__details--subtitle"
    >
      undefined undefined undefined minutos
    </p>
  </div>
</div>
`;
